📖 [amp story shopping] documentation (#37987)
ampproject/amphtml: extensions/amp-story-shopping/amp-story-shopping.md

<amp-img alt="An example of amp story shopping tag in each type of configuration" src="https://user-images.githubusercontent.com/3860311/152416815-61a63a88-76c0-4ae8-9300-a7ad277dbe7a.jpg" layout="intrinsic" width="806" height="428">

The `amp-story-shopping` component creates configurable, templated shopping experiences within [amp-story](https://amp.dev/documentation/components/amp-story/?format=stories).

Specify a shopping experience by defining one `amp-story-shopping-attachment` and one or more `amp-story-shopping-tag` elements for each product on the [`amp-story-page`](https://amp.dev/documentation/components/amp-story-page/?format=stories).

## amp-story-shopping-tag

<amp-img alt="An example of amp story shopping tag in each type of configuration" src="https://user-images.githubusercontent.com/3860311/155754637-3403a9dd-c4c9-44f3-ad6d-e4d166d30ad2.gif" layout="intrinsic" width="844" height="102">

Use `amp-story-shopping-tag` elements to indicate shop-able elements within [`amp-story-page`](https://amp.dev/documentation/components/amp-story-page/?format=stories).
Tapping them opens a product description page (PDP) within the `amp-story-shopping-attachment`.

> -   They must be a child of [`amp-story-grid-layer`](https://amp.dev/documentation/components/amp-story-grid-layer/).

Reviewer: This list needs an introduction of what it is a list of. Maybe something like: `To use `amp-story-shopping-tag` it must meet the following requirements:` - and then remove the `They must`/`they are` from the first two points.

-   They are positioned [`absolute`](https://developer.mozilla.org/en-US/docs/Web/CSS/position#values). Use custom CSS to place them on the page.
-   Use `top` and `left` CSS rules with percentage based units for responsive layout.

```html
...
<style amp-custom>
  [data-product-id="product1"] {
    top: 30%;
    left: 30%;
  }
</style>
...
<amp-story-grid-layer template="vertical">
  <amp-story-shopping-tag data-product-id="product1"></amp-story-shopping-tag>
</amp-story-grid-layer>
```

> Example of positioning a amp-story-shopping-tag and testing responsiveness:

Reviewer: All instances of `amp-story-shopping-tag` should be wrapped in back ticks

<amp-img alt="An example of positioning an amp story shopping tag and teseting responsiveness" src="https://user-images.githubusercontent.com/3860311/155751130-558b5ab6-1db3-4ca7-b913-4be4761fdb29.gif" layout="intrinsic" width="840" height="543">

> ### Attributes

Reviewer: Are these attributes specific to `amp-story-shopping-tag`?

#### `data-product-id` {string} required

> Used to associate the `amp-story-shopping-tag` with product data.
> It should be equal to the `productId` value of the associated product's JSON data.

Reviewer: ```suggestion
Associates the `amp-story-shopping-tag` with product data.
Must be equal to the `productId` value of the product's JSON data.
```
